Models/BookModel.php

```diff
@@ -12,37 +12,36 @@ public function __construct()
     parent::__construct();
   }
 
-
   /**
    * Get a book by id.
    *
    * @param int $id
    *
    * @return BookEntity|null
    */
-  public function get($id)
+  function get($id)
   {
-    $sql = "SELECT * FROM books WHERE id = ?";
-    $data = array($id);
-    $statement = $this->executeStatement($sql, $data);
-    $row = $statement->fetchObject();
+    $query = $this->dsql_connection->dsql();
+    $result = $query->table('books')
+      ->where('id', '=', $id)
+      ->getRow();
     $book = NULL;
-    if ($row) {
+    if ($result) {
       $book = new BookEntity(array(
-        'id' => $row->id,
-        'title' => $row->title,
-        'description' => $row->description,
-        'rating' => $row->rating,
-        'ISBN_13' => $row->ISBN_13,
-        'ISBN_10' => $row->ISBN_10,
-        'image' => $row->image,
-        'language' => $row->language,
-        'price' => $row->price,
-        'currency' => $row->currency,
-        'buy_link' => $row->buy_link
+        "id" => $result['id'],
+        "title" => $result['title'],
+        "description" => $result['description'],
+        "rating" => $result['rating'],
+        "ISBN_13" => $result['ISBN_13'],
+        "ISBN_10" => $result['ISBN_10'],
+        "image" => $result['image'],
+        "language" => $result['language'],
+        "price" => $result['price'],
+        "currency" => $result['currency'],
+        "buy_link" => $result['buy_link']
       ));
+      return $book;
```

Comment:
This line is not correctly placed. If the query does not return anything, this line is not executed.